dart_application_2/bin/task.dart

```diff
@@ -0,0 +1,212 @@
+      print("enter city name");
+      city = stdin.readLineSync()!;
+      exceptionFlag = ValidationsUtil.validateString(city);
+    } while (exceptionFlag);
+
+    do {
+      exceptionFlag = false;
+      print("enter district name");
+      district = stdin.readLineSync()!;
+      exceptionFlag = ValidationsUtil.validateString(district);
+    } while (exceptionFlag);
+    do {
+      exceptionFlag = false;
+      print("enter state name");
+      state = stdin.readLineSync()!;
+      exceptionFlag = ValidationsUtil.validateString(state);
+    } while (exceptionFlag);
+
+    Address address = new Address(
+        streetNumber: streetNumber,
+        city: city,
+        district: district,
+        state: state);
+    do {
+      exceptionFlag = false;
+      print("enter roll number");
+      rollString = stdin.readLineSync()!;
+      exceptionFlag = ValidationsUtil.validateNumber(rollString);
+     //         TODO: check if the student corresponding to this roll number already exists
+    } while (exceptionFlag);
+    
+    rollNumber = int.parse(rollString);
+
+    Set<Course> courseSet = {};
+
+    while (courseSet.length < 4) {
+      exceptionFlag = false;
+      print("chose 4 courses from 0,1,2,3,4,5");
+      courseString = stdin.readLineSync()!;
+      exceptionFlag = ValidationsUtil.validateOption(courseString, 0, 5);
+      if (exceptionFlag) continue;
+      int courseNumber = int.parse(courseString);
+      courseSet.add(Course.values[courseNumber]);
+    }
+        
+
+    Student student = new Student(
+        fullName: fullName,
+        age: age,
+        address: address,
+        rollNumber: rollNumber,
+        courseSet: courseSet);
+    return student;
+  }
+
+  /*  displays details of students based on sort parameter
+    */
+  static _displayDetailsOption(List<Student> studentList) {
+    studentList = SortItems.sortItems(studentList);
+    studentList.forEach((student) {
+      print("${student.fullName}" +
+          " ${student.rollNumber}  ${student.age})" +
+          " ${student.address.streetNumber}" +
+          "${student.address.city} ${student.address.state}" +
+          "${student.courseSet}");
+    });
+  }
+
+  /**
+   * delete details of student based on given roll
+   */
+  static List<Student> _deleteDetailsOption(List<Student> studentList) {
+    String rollNumberString;
+    int rollNumber, indexRollNumber;
+    bool exceptionFlag;
+    if (studentList.isEmpty) {
+      print("No element in the list");
+      return studentList;
+    }
+    do {
+      do {
+        exceptionFlag = false;
+        print("enter roll number to be deleted");
+        rollNumberString = stdin.readLineSync()!;
+        exceptionFlag = ValidationsUtil.validateNumber(rollNumberString);
+      } while (exceptionFlag);
+      rollNumber = int.parse(rollNumberString);
+      indexRollNumber = studentList
+          .indexWhere(((student) => student.rollNumber == rollNumber));
+    } while (indexRollNumber == -1);
```

Comment:
What is this while and its condition for ? 
You've handled the validation in the inner while 
Is this while required ? 
Also not required to do indexWhere and removeAt 
Checkout list.removeWhere, it does what you need in one step

Reply:
The inner while only checks if the given roll number is passing the validation criteria of roll number, it does not check if that roll number exist in the list of students, the indexWhere checks if that roll number exist or not in the list, if that index is not in the list then another input for roll number will be asked again and validated again. The loop exits only when a roll number present in the list is entered and then details of that corresponding roll number is deleted using removeWhere.